traveler-swift-core/TravelerKit/Traveler.swift

```diff
@@ -73,15 +73,23 @@ public class Traveler {
         OperationQueue.main.addOperation(blockOperation)
     }
 
-    func fetchCatalogItemDetails(_ catalogItem: Product, completion: @escaping (CatalogItemDetails?, Error?) -> Void) {
-        let fetchOperation: AuthenticatedRemoteFetchOperation<CatalogItemDetails>
-        if let travelerProfileId = session.identity {
-            fetchOperation = AuthenticatedRemoteFetchOperation<CatalogItemDetails>(path: .catalogItem(catalogItem, travelerId: travelerProfileId), session: session)
-        } else {
-            fetchOperation = AuthenticatedRemoteFetchOperation<CatalogItemDetails>(path: .catalogItem(catalogItem, travelerId: nil), session: session)
+    func fetchProductDetails(_ product: Product, completion: @escaping (CatalogItemDetails?, Error?) -> Void) {
```

Comment:
This is just to try and play with...might be fun

`func fetchCatalogItemDetails<T : ProductItem>(_ item: T, completion: @escaping (T.Details?, Error?) -> Void)`

Reply:
Did this not work?